### PR TITLE
Update source-ip doc with new ESIPP fields

### LINKED ISSUE #4271: Source IP tutorial uses outdated (since 1.7) annotation method
**This is a...** 
- [x] Bug Report

**Problem:**
The method for specifying traffic should be kept local to the node has been promoted from an annotation to a field in 1.7.

https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#feature-availability explains this.

**Proposed Solution:**
Update this page to match.

I'm not sure what our policy is on documenting previous versions.  Do we assume everyone is running 1.7 and only document the new method?

**Page to Update:**
k8s.io/docs/tutorials/services/source-ip/

### PR DESCRIPTION
**Pull request by MrHohn**
Fixes #4271.

/assign @ahmetb 
cc @craigbox @bowei

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4279)

### CODE REVIEW

File: docs/tutorials/services/source-ip.md [hunk 1]
```diff
@@ -157,19 +157,20 @@ Visually:
...
 
-To avoid this, Kubernetes 1.5 has a beta feature triggered by the
-`service.beta.kubernetes.io/external-traffic` [annotation](/docs/user-guide/load-balancer/#loss-of-client-source-ip-for-external-traffic).
-Setting it to the value `OnlyLocal` will only proxy requests to local endpoints,
-never forwarding traffic to other nodes and thereby preserving the original
-source IP address. If there are no local endpoints, packets sent to the node
-are dropped, so you can rely on the correct source-ip in any packet processing
-rules you might apply a packet that make it through to the endpoint.
+To avoid this, Kubernetes has a feature to preserve the client source ip
+[check here for feature availability](/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip).
```
Comment (ahmetb) [inline docs/tutorials/services/source-ip.md]: > check here for feature availability

missing parens around here

File: docs/tutorials/services/source-ip.md [hunk 2]
```diff
@@ -157,19 +157,20 @@ Visually:
...
+Setting `service.spec.externalTrafficPolicy` to the value `Local` will only
+proxy requests to local endpoints, never forwarding traffic to other nodes
+and thereby preserving the original source IP address. If there are no
+local endpoints, packets sent to the node are dropped, so you can rely
+on the correct source-ip in any packet processing rules you might apply a
+packet that make it through to the endpoint.
 
-Set the annotation as follows:
+Set `service.spec.externalTrafficPolicy` field as follows:
```
Comment (bowei) [inline docs/tutorials/services/source-ip.md]: Set *the* `service...` field

File: docs/tutorials/services/source-ip.md [hunk 3]
```diff
@@ -157,19 +157,20 @@ Visually:
...
 
 
-To avoid this, Kubernetes 1.5 has a beta feature triggered by the
-`service.beta.kubernetes.io/external-traffic` [annotation](/docs/user-guide/load-balancer/#loss-of-client-source-ip-for-external-traffic).
-Setting it to the value `OnlyLocal` will only proxy requests to local endpoints,
-never forwarding traffic to other nodes and thereby preserving the original
-source IP address. If there are no local endpoints, packets sent to the node
-are dropped, so you can rely on the correct source-ip in any packet processing
-rules you might apply a packet that make it through to the endpoint.
+To avoid this, Kubernetes has a feature to preserve the client source ip
```
Comment (ahmetb) [inline docs/tutorials/services/source-ip.md]: ip &rarr; IP

File: docs/tutorials/services/source-ip.md [hunk 4]
```diff
@@ -230,8 +231,8 @@ client_address=10.240.0.5
 ...
 ```
 
-However, if you're running on GKE/GCE, setting the same `service.beta.kubernetes.io/external-traffic`
-annotation to `OnlyLocal` forces nodes *without* Service endpoints to remove
+However, if you're running on GKE/GCE, setting the same `service.spec.externalTrafficPolicy`
+field to `Local` forces nodes *without* Service endpoints to remove
 themselves from the list of nodes eligible for loadbalanced traffic by
 deliberately failing health checks. We expect to roll this feature out across a
```
Comment (bowei) [inline docs/tutorials/services/source-ip.md]: Remove sentence starting with "We expect..."

File: docs/tutorials/services/source-ip.md (hunk 2 above)
Comment (MrHohn) [inline docs/tutorials/services/source-ip.md]: Done.

File: docs/tutorials/services/source-ip.md (hunk 4 above)
Comment (MrHohn) [inline docs/tutorials/services/source-ip.md]: Done.

File: docs/tutorials/services/source-ip.md (hunk 3 above)
Comment (MrHohn) [inline docs/tutorials/services/source-ip.md]: Done.

File: docs/tutorials/services/source-ip.md (hunk 1 above)
Comment (MrHohn) [inline docs/tutorials/services/source-ip.md]: Added parens.

**Comment by MrHohn** [comment]
Thanks for the quick review!

**Comment by craigbox** [comment]
Indeed. You guys are great.

**Pull request merged**